app/src/main/java/org/p2p/wallet/common/ui/widget/EarnWidget.kt

```diff
@@ -0,0 +1,183 @@
+                textViewEarnMessage.setText(state.messageTextRes)
+                buttonEarn.setText(state.buttonTextRes)
+            }
+        }
+    }
+
+    private fun makeAlignStartContent() = with(binding) {
+        textViewEarnTitle.apply {
+            textAlignment = TEXT_ALIGNMENT_TEXT_START
+            setTextAppearance(R.style.UiKit_TextAppearance_Regular_Text3)
+        }
+        val horizontalPadding = SMALL_BUTTON_TEXT_PADDING_DP.toPx()
+        buttonEarn.apply {
+            textAlignment = TEXT_ALIGNMENT_TEXT_START
+            setPadding(horizontalPadding, 0, horizontalPadding, 0)
+        }
+    }
+
+    private fun makeAlignCenterContent() = with(binding) {
+        textViewEarnTitle.apply {
+            textAlignment = TEXT_ALIGNMENT_CENTER
+            setTextAppearance(R.style.UiKit_TextAppearance_SemiBold_Text1)
+        }
+        val horizontalPadding = DEFAULT_BUTTON_TEXT_PADDING_DP.toPx()
+        buttonEarn.apply {
+            textAlignment = TEXT_ALIGNMENT_CENTER
+            setPadding(horizontalPadding, 0, horizontalPadding, 0)
+        }
+    }
+
+    fun setDepositTokens(tokens: List<String>) {
+        val container = binding.viewTokenContainer
+        if (container.childCount < tokens.size) {
+            for (i in container.childCount until tokens.size) {
+                val binding = ItemDepositTokenBinding.inflate(
+                    LayoutInflater.from(context), container, true
+                ).apply {
+                    val params = tokenImageView.layoutParams as MarginLayoutParams
+                    params.marginEnd = (TOKEN_MARGIN_END_DP * i).toPx()
+                    tokenImageView.layoutParams = params
+                }
+                val holder = DepositTokenViewHolder(binding, glideManager)
+                binding.root.tag = holder
+            }
+        } else if (container.childCount > tokens.size) {
+            if (tokens.isEmpty()) {
+                removeAllViews()
+            } else {
+                container.removeViewsInLayout(
+                    tokens.size, container.childCount - tokens.size
+                )
+            }
+        }
+        for (i in 0 until container.childCount) {
+            (container.getChildAt(i)?.tag as? DepositTokenViewHolder)?.apply {
+                bind(tokens[i])
+            }
+        }
+    }
+}
+
+class DepositTokenViewHolder(
+    private val binding: ItemDepositTokenBinding,
+    private val glideManager: GlideManager
+) {
+
+    companion object {
+        private const val IMAGE_SIZE = 16
+    }
+
+    fun bind(tokenUrl: String) = with(binding) {
+        glideManager.load(tokenImageView, tokenUrl, IMAGE_SIZE)
+    }
+}
+
+fun BigDecimal.formatTicker(): String = this.run {
+    if (isZero()) this.toString() else DecimalFormatter.format(this, TICKER_DECIMALS)
+}
+
+sealed class EarnWidgetState(@ColorRes val backgroundColor: Int = R.color.bg_rain) {
```

Comment:
to a separate file or inside the Widget class

Reply:
separate file please